src/cocotb/share/lib/embed/gpi_embed.cpp

```diff
@@ -247,7 +246,7 @@ extern "C" COCOTB_EXPORT void _embed_sim_cleanup(void) {
     if (Py_IsInitialized()) {
         to_python();
         PyGILState_Ensure();  // Don't save state as we are calling Py_Finalize
-        Py_DecRef(pEventFn);
+        Py_XDECREF(pEventFn);
```

Comment:
if `set_sim_event_callback()` is optional, `pEventFn` needs to be null checked here and where it's called.

Reply:
XDECREF includes a null check. But yes, the other location needs it as well.

Reply:
Looks like it's already checked where it's called as well.